src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -444,7 +444,12 @@ unique_ptr<CatalogEntry> TableCatalogEntry::SetDefault(ClientContext &context, S
 		auto copy = columns[i].Copy();
 		if (default_idx == i) {
 			// set the default value of this column
-			D_ASSERT(!copy.Generated()); // Shouldnt reach here - DEFAULT value isn't supported for Generated Columns
+			if (copy.Generated()) {
+				throw BinderException("DEFAULT values are not supported for generated columns!");
+			}
+			if (copy.IsSerial()) {
+				throw BinderException("DEFAULT values are not supported for SERIAL type!");
```

Comment:
this is triggered in the case we try adding a default value in an ALTER TABLE statement

Reply:
I think we want to catch this earlier, in the transformer, if possible
That's why the assertion was there, because it should already be handled in the transformer

Reply:
sure but it's not being caught xD I think for the time being it is better to have this in backlog and pray that my implementation works, will get to it when I am done :)